python/lsst/pex/config/configDictField.py

```diff
@@ -48,6 +48,7 @@ def __setitem__(self, k, x, at=None, label="setitem", setHistory=True):
             raise FieldValidationError(self._field, self._config, msg)
 
         # validate keytype
+        k = _autocast(k, self._field.keytype)
```

Comment:
@r-owen it's probably the case that we should be using `_autocast` more consistently in `pex_config`. This works because a `future` `str` is an instance of `unicode`, although at present, you end up with the unicode-ness being stripped so this is going to fail (and `pex_config` will fail in general) if we pass in unicode strings that can't be converted to native Python 2 `str`. There are advantages to dropping Python 2.